packages/nextcloud/lib/src/helpers/spreed.dart

```diff
@@ -0,0 +1,199 @@
+    final version = capabilities.capabilities.spreedPublicCapabilities?.spreed.version;
+    return (
+      version != null && Version.parse(version).major == supportedVersion,
+      supportedVersion,
+    );
+  }
+}
+
+/// Conversation types.
+///
+/// Use [value] to get the integer representation that is used in the API.
+/// See https://github.com/nextcloud/spreed/blob/master/lib/Room.php.
+enum RoomType {
+  /// "One to one".
+  oneToOne,
+
+  /// Group.
+  group,
+
+  /// Public.
+  public,
+
+  /// Changelog.
+  changelog,
+
+  /// Former "One to one" (When a user is deleted from the server or removed from all their conversations, `1` "One to one" rooms are converted to this type).
+  oneToOneFormer,
+
+  /// Note to self.
+  noteToSelf;
+
+  /// Integer representation of the [ParticipantType].
+  int get value => index + 1;
+}
+
+/// Types of chat messages.
+///
+/// Use [name] to get the string representation that is used in the API.
+/// See https://github.com/nextcloud/spreed/blob/master/lib/Chat/ChatManager.php.
+enum MessageType {
+  /// Message.
+  comment,
+
+  /// Message from the system.
+  system,
+
+  /// An object shared to the room.
+  // ignore: constant_identifier_names
+  object_shared,
+
+  /// Message from a command.
+  command,
+
+  /// Deleted message.
+  // ignore: constant_identifier_names
+  comment_deleted,
+
+  /// Emoji reaction.
+  reaction,
+
+  /// Deleted emoji reaction.
+  // ignore: constant_identifier_names
+  reaction_deleted;
+}
+
+/// Actor types of chat messages.
+///
+/// Use [name] to get the string representation that is used in the API.
+/// See https://github.com/nextcloud/spreed/blob/master/lib/Model/Attendee.php.
+enum ActorType {
+  /// Logged-in users.
+  users,
+
+  /// Groups.
+  groups,
+
+  /// Guest users.
+  guests,
+
+  /// E-mails.
+  emails,
+
+  /// Circles.
+  circles,
+
+  /// Users whose messages are bridged in by the Matterbridge integration.
+  bridged,
+
+  /// Used by commands and the changelog conversation.
+  bots,
+
+  /// Users from other instances.
+  // ignore: constant_identifier_names
+  federated_users,
+
+  /// Users from SIP.
+  phones;
+}
+
+/// Participant types.
+///
+/// Use [value] to get the integer representation that is used in the API.
+/// See https://github.com/nextcloud/spreed/blob/master/lib/Participant.php.
+enum ParticipantType {
+  /// Owner.
+  owner,
+
+  /// Moderator.
+  moderator,
+
+  /// User.
+  user,
+
+  /// Guest.
+  guest,
+
+  /// User following a public link.
+  userFollowingPublicLink,
+
+  /// Guest with moderator permissions.
+  guestWithModeratorPermissions;
+
+  /// Integer representation of the [ParticipantType].
+  int get value => index + 1;
+}
+
+/// Attendee permissions.
+///
+/// Use [value] to get the integer representation that is used in the API.
+/// See https://github.com/nextcloud/spreed/blob/master/lib/Model/Attendee.php.
+enum ParticipantPermission {
```

Comment:
Please mention and explain both the `fromValue` method and the `ParticipantPermissionsValue` extension,how and why they are used.